compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -577,12 +579,16 @@ public BLangNode transform(ModulePartNode modulePart) {
         compilationUnit.name = currentCompUnitName;
         compilationUnit.setPackageID(packageID);
         Location pos = getPosition(modulePart);
-
+        String previousPkgID = this.packageID.name.value;
         // Generate import declarations
         for (ImportDeclarationNode importDecl : modulePart.imports()) {
             BLangImportPackage bLangImport = (BLangImportPackage) importDecl.apply(this);
             bLangImport.compUnit = this.createIdentifier(pos, compilationUnit.getName());
             compilationUnit.addTopLevelNode(bLangImport);
+            if (this.packageID.name.value != previousPkgID) {
```

Comment:
You can't use `==` with strings, need to use `.equals()`, and we can just compare `name` field I think the equals method is overridden there.
I'm not sure it `this.pakcageID` get changed here.

Reply:
I think we can clear the `constantSet` at the end of this method.

Reply:
Yes. I'll change it.